src/components/ToDoComponent/Card.tsx

```diff
@@ -0,0 +1,112 @@
+import { Image, Text, View, TouchableOpacity } from 'react-native';
+import React, { useState } from 'react';
+import Animated, {
+  runOnJS,
+  useAnimatedGestureHandler,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+import {
+  GestureHandlerRootView,
+  PanGestureHandler,
+} from 'react-native-gesture-handler';
+import Images from '../../constants/images/Image';
+import Toast from 'react-native-toast-message';
+import { CardStyles } from './Styles/CardStyles';
+
+const Card = ({ item, posStyle, changecard, removeCard }: any) => {
+  const translateY = useSharedValue(0);
+  const [checked, setChecked] = useState(false);
+  let deleteCard = 'false';
+
+  const panGesture = useAnimatedGestureHandler({
+    onActive: (event) => {
+      if (translateY.value < 150) {
+        translateY.value = event.translationY;
+      }
+    },
+    onEnd: () => {
+      translateY.value = withTiming(0, {
+        duration: 700,
+      });
+      runOnJS(changecard)(item.id);
+    },
+  });
+
+  const animatedStyle = useAnimatedStyle(() => ({
+    transform: [
+      {
+        translateY: translateY.value,
+      },
+    ],
+  }));
+
+  const markDone = () => {
+    deleteCard = 'true';
+    setChecked(true);
+    Toast.show({
+      type: 'error',
+      text1: 'Mark as done',
+      text2: 'Click here to mark undone',
+      position: 'bottom',
+      bottomOffset: 80,
+      onPress: () => {
+        Toast.hide();
+        deleteCard = 'false';
+        setChecked(false);
+      },
+    });
+    setTimeout(() => {
```

Comment:
Is it a good idea to use setTimeout in a function directly, and not in a custom hook or a useEffect where it can also be cleared?

Reply:
I will do that